### PR TITLE
DOCS-1191: Updating SSE params, general fixups

### LINKED ISSUE #1191: [RELEASE] MC RELEASE.2024-04-18T16-45-29Z
PRODUCT [RELEASE.2024-04-18T16-45-29Z](https://github.com/minio/mc/releases/tag/RELEASE.2024-04-18T16-45-29Z) has changes that impact the docs:

- [x] major changes to SSE-encryption settings, do a full pass of all commands
- [x] New command `mc idp ldap accesskey create-with-login` [4894](https://github.com/minio/mc/pull/4894)
- [x] `mc admin bucket` is now hidden, may need to mark as deprecated the entire tree here [4905](https://github.com/minio/mc/pull/4905)

### PR DESCRIPTION
**Pull request by ravindk89**
Closes #1191 

# Summary

Finally getting around to this mc release

- Added docs for enc-c, enc-s3, enc-kms
- Some docs are making assumptions around behavior that needs to be fixed _first_
- Drive-by linker fixes

Staged: http://192.241.195.202:9000/staging/DOCS-1191/linux/index.html

### CODE REVIEW

File: source/reference/minio-mc/mc-cat.rst [hunk 1]
```diff
@@ -105,22 +105,84 @@ Parameters
       :start-after: start-version-id-desc
       :end-before: end-version-id-desc
 
-.. mc-cmd:: --encrypt-key
+.. mc-cmd:: --enc-kms
+
+   Encrypt or decrypt objects using server-side :ref:`SSE-KMS encryption <minio-sse>` with client-managed keys.
+   
+   The parameter accepts a key-value pair formatted as ``KEY=VALUE``
```
Comment (ravindk89) [inline source/reference/minio-mc/mc-cat.rst:112]: note: Assuming @zveinn can identify the root issue w.r.t. the key parser clipping padding bytes.

Also could not figure out how to specify HEX keys, if I can get an example I'll patch those in.

File: source/reference/minio-mc/mc-cat.rst [hunk 2]
```diff
@@ -105,22 +105,84 @@ Parameters
...
-   Encrypt or decrypt objects using server-side encryption with
-   client-specified keys. Specify key-value pairs as ``KEY=VALUE``.
+   Encrypt or decrypt objects using server-side :ref:`SSE-S3 encryption <minio-sse>` with KMS-managed keys.
+   Specify the full path to the object as ``alias/bucket/prefix/object``.
+
+   For example:
+
+   .. code-block:: shell
+
+      --enc-s3 "myminio/mybucket/prefix/object.obj"
```
Comment (ravindk89) [inline source/reference/minio-mc/mc-cat.rst:144]: There's a newer PR that ensures this works as expected, no need to mention it afaik.

File: source/reference/minio-mc/mc-cat.rst [hunk 3]
```diff
@@ -105,22 +105,84 @@ Parameters
...
+   The parameter accepts a key-value pair formatted as ``KEY=VALUE``
+
+   - ``KEY`` must contain the full path to the object as ``alias/bucket/path/object``.
+   - ``VALUE`` must contain the 32-byte Base64-encoded data key to use for encrypting object(s).
+
+   For example:
+
+   .. code-block:: shell
+
+      --enc-c "myminio/mybucket/prefix/object.obj=bXktc3NlLWMta2V5Cg=="
```
Comment (ravindk89) [inline source/reference/minio-mc/mc-cat.rst:172]: Note, ATM this fails due to the parsing doing some strange indexing on the buffer bytes, needs a fix.

File: source/reference/minio-mc/mc-idp-ldap-accesskey-create-with-login.rst [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+.. mc:: mc idp ldap accesskey create-with-login
+
+.. versionadded:: mc RELEASE.2024-04-18T16-45-29Z
+
+Description
+-----------
+
+.. start-mc-idp-ldap-accesskey-create-desc
+
+The :mc:`mc idp ldap accesskey create-with-login` uses interactive terminal-based login in prompt to authenticate with the external AD/LDAP server and generate access keys for use with MinIO.
```
Comment (feorlen) [inline source/reference/minio-mc/mc-idp-ldap-accesskey-create-with-login.rst:23]: Is `in prompt` supposed to be here?

File: source/reference/minio-mc/mc-idp-ldap-accesskey-create-with-login.rst [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+   A secret key to use once successfully authenticated.
+   Omit to let MinIO randomly generate a value.
+
+   Requires :mc-cmd:`~mc idp ldap accesskey create-with-login --access-key`
+
+.. mc-cmd:: --policy
+   :optional:
+
+   File path to the JSON-formatted :ref:`policy <minio-policy>` to use for the account.
+   This policy can only further limit those privileges associated with the authenticated AD/LDAP user.
```
Comment (feorlen) [inline source/reference/minio-mc/mc-idp-ldap-accesskey-create-with-login.rst:99]: an idea
```suggestion
   This policy _cannot_ grant additional privileges beyond the privileges associated with the authenticated AD/LDAP user.
```

File: source/reference/minio-mc/mc-idp-ldap-accesskey-create-with-login.rst (hunk 4 above)
Comment (ravindk89) [reply source/reference/minio-mc/mc-idp-ldap-accesskey-create-with-login.rst:23]: nope, will fix

File: source/reference/minio-mc/mc-idp-ldap-accesskey-create-with-login.rst [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+.. mc-cmd:: --description
+   :optional:
+
+   Add a description for the service account.
+   For example, you might specify the reason the access key exists.
+
+.. mc-cmd:: --expiry-duration
+   :optional:
+
+   Length of time the access key pair should remain valid for use in ``#d#h#s`` format.
```
Comment (feorlen) [inline source/reference/minio-mc/mc-idp-ldap-accesskey-create-with-login.rst:117]: A question about the behavior: you can specify some combination of days, hours, or seconds, but not _minutes_?

File: source/reference/minio-mc/mc-cat.rst (hunk 3 above)
Comment (ravindk89) [reply source/reference/minio-mc/mc-cat.rst:172]: Figured it out, we have to drop all `==` trailing buffers to make it work.

File: tee [hunk 7]
```diff
@@ -0,0 +1 @@
+/dev/urandom foo.txt
```
Comment (feorlen) [inline tee:1]: Is this used somewhere?

**Comment by feorlen** [comment]
An existing issue in the original: this doesn't appear in the TOC anywhere? 
`source/reference/minio-mc/mc-idp-ldap-accesskey-create.rst`

File: source/reference/minio-mc/mc-idp-ldap-accesskey-create-with-login.rst (hunk 6 above)
Comment (ravindk89) [reply source/reference/minio-mc/mc-idp-ldap-accesskey-create-with-login.rst:117]: I am not sure - will have to check and maybe patch that later

File: tee (hunk 7 above)
Comment (ravindk89) [reply tee:1]: huh that's where that went.

**Review by djwfyi — APPROVED** [review]
Just some nits.
Nothing major that I saw, other than I was briefly confused about the difference between SSE-KMS and SSE-C. I think that's clarified in other docs that you link to. You might consider adding a little blurb to the include file, but not critical.

Otherwise, LGTM.

File: source/includes/common-minio-sse.rst [hunk 8]
```diff
@@ -23,4 +23,115 @@ MinIO generates a random 256-bit unique Object Encryption Key (OEK) and uses tha
...
+
+      --enc-kms "myminio/mybucket/prefix/object.obj=mybucketencryptionkey"
+
+   You can specify multiple encryption keys by repeating the parameter.
+
+   Specify the path to a prefix to apply encryption to all matching objects at that path:
+
+   .. code-block:: shell
+
+      --enc-kms "myminio/mybucket/prefix/mybucketencryptionkey"
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/includes/common-minio-sse.rst:63]: Yeah definitely want teh equal sign in there

File: source/includes/common-minio-sse.rst [hunk 9]
```diff
@@ -23,4 +23,115 @@ MinIO generates a random 256-bit unique Object Encryption Key (OEK) and uses tha
...
+          Omit the padding or use a Base64 encoder that supports RAW formatting.
+
+   - ``KEY`` - the full path to the object as ``alias/bucket/path/object``.
+   - ``VALUE`` - the 32-byte RAW Base64-encoded data key to use for encrypting object(s).
+
+   For example:
+
+   .. code-block:: shell
+
+      # RawBase64-Encoded string "mybucket32byteecryptionkeyssec"
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/includes/common-minio-sse.rst:121]: definitely. I got caught by the double 'e's

File: source/reference/minio-mc/mc-cat.rst [hunk 10]
```diff
@@ -91,36 +94,41 @@ Parameters
...
-   - Each ``VALUE`` represents the data key to use for encrypting 
-      object(s).
-
-   Enclose the entire list of key-value pairs passed to 
-   :mc-cmd:`~mc cat --encrypt-key` in double quotes ``"``.
-
-   :mc-cmd:`~mc cat --encrypt-key` can use the ``MC_ENCRYPT_KEY``
-   environment variable for retrieving a list of encryption key-value pairs
-   as an alternative to specifying them on the command line.
+   Extracts from a remote zip file.
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/reference/minio-mc/mc-cat.rst:130]:
```suggestion
   Extracts the contents from a zip file on the source to the remote.
```

File: source/reference/minio-mc/mc-idp-ldap-accesskey-create-with-login.rst [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+.. code-block:: shell
+   :class: copyable
+
+   mc idp ldap accesskey create-with-login https://minio.example.net --expiry-duration 24h
+
+Create a new access-key pair that expires after a date
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following command creates a new access key pair to use with the currently authenticated user on the ``minio`` alias.
+The credentials expire after February 29, 2024.
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/reference/minio-mc/mc-idp-ldap-accesskey-create-with-login.rst:193]: I've always found the linear nature of time frustrating.

File: source/reference/minio-mc/mc-idp-ldap-accesskey-create.rst [hunk 12]
```diff
@@ -49,8 +49,7 @@ The :mc:`mc idp ldap accesskey create` allows you to add LDAP access key pairs.
                                           [--policy <value>]          \
                                           [--name <value>]            \
                                           [--description <value>]     \
-                                          [--expiry-duration <value>] \ 
-                                          [--login <site>]
+                                          [--expiry <value>]          \                                         [--expiry-duration <value>]
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/reference/minio-mc/mc-idp-ldap-accesskey-create.rst:52]: whoops yes.

**Comment by ravindk89** [comment]
@zveinn if there's nothing obviously wrong here I'll merge this up. Thanks!

**Pull request merged**